lib/webhdfs/client_v1.rb

```diff
@@ -59,32 +61,62 @@ def initialize(host='localhost', port=50070, username=nil, doas=nil, proxy_addre
 
       @kerberos = false
       @kerberos_keytab = nil
+      @renew_kerberos_delegation_token_time_hour = renew_kerberos_delegation_token_time_hour
+      @kerberos_delegation_token = nil
+      @kerberos_token_updated_at = Time.now
       @http_headers = http_headers
     end
 
+    def should_kerberos_token_updated?
+      @kerberos_token_updated_at + (@renew_kerberos_delegation_token_time_hour * 60 * 60) <= Time.now
+    end
+
+
+    def get_cached_kerberos_delegation_token
+      return @kerberos_delegation_token if !@kerberos_delegation_token.nil? && !should_kerberos_token_updated?
+
+      if !@kerberos_delegation_token
+        @kerberos_delegation_token = get_kerberos_delegation_token(@username)
+      else
+        renew_kerberos_delegation_token(@kerberos_delegation_token)
+      end
+      @kerberos_token_updated_at = Time.now
+      @kerberos_delegation_token
+    rescue => e
+      raise WebHDFS::RequestFailedError, e.message
+    end
+
     # curl -i -X PUT "http://<HOST>:<PORT>/webhdfs/v1/<PATH>?op=CREATE
     #                 [&overwrite=<true|false>][&blocksize=<LONG>][&replication=<SHORT>]
     #                 [&permission=<OCTAL>][&buffersize=<INT>]"
+    #                 [&delegation=<DELEGATION_TOKEN>]"
     def create(path, body, options={})
       if @httpfs_mode
         options = options.merge({'data' => 'true'})
       end
+      if @renew_kerberos_delegation_token_time_hour
+        options = options.merge('delegation' => get_cached_delegation_token)
```

Comment:
The method name is `get_cached_kerberos_delegation_token` but this caller is still not updated.